features/features.go

```diff
@@ -220,3 +234,32 @@ func Flag(tr confGetter, feature SnapdFeature) (bool, error) {
 	}
 	return false, fmt.Errorf("%s can only be set to 'true' or 'false', got %q", feature, isEnabled)
 }
+
+type FeatureInfo struct {
+	Supported bool `json:"supported"`
+	Enabled   bool `json:"enabled"`
+}
+
+func All(tr confGetter) map[string]FeatureInfo {
+	knownFeatures := KnownFeatures()
+	allFeaturesInfo := make(map[string]FeatureInfo, len(knownFeatures))
+	for _, feature := range knownFeatures {
+		enabled, err := Flag(tr, feature)
+		if err != nil {
+			// Skip features with values other than true or false
+			continue
+		}
+		// Features implicitly supported if no callback exists
+		supported := true
+		if callback, exists := featuresSupportedCallbacks[feature]; exists {
+			supported = callback()
```

Comment:
in which case the callback would need to return more than a bool

Reply:
It could either return `(bool, error)` where a `false` value must always be accompanied by a non-`nil` error. Or we could simply return `error` where `nil` means supported and non-`nil` means supported. The former is redundant but clearer, while the latter is not redundant (and matches usage of `Flag()`), but is slightly less clear on its own. Any preference between these two approaches?

Reply:
I've implemented `(bool, error)` for now. Feel free to suggest otherwise if you'd prefer.

Reply:
Actually, `(bool, string)`, since it's really a description rather than an error.